integration_tests/test_domestic_migration.py

```diff
@@ -239,32 +252,33 @@ def test_addresses_during_moves(simulants_on_adjacent_timesteps, unit_id_col, ad
         after_units = after_units.reindex(total_index)
         mask_moved_units = before_units[address_id_col] != after_units[address_id_col]
 
-        # check that the number of moved units that have the same details is very low
+        # Check that the number of moved units that have the same details is very low
         # NOTE: we cannot assert that all moved units have different details
-        # because they are free to move within the same state or even puma
+        # because they are free to move within the same state or even PUMA
         assert (
             (
-                before_units.loc[mask_moved_units, other_address_cols]
-                == after_units.loc[mask_moved_units, other_address_cols]
+                before_units.loc[mask_moved_units, address_cols]
+                == after_units.loc[mask_moved_units, address_cols]
             )
             .all(axis=1)
             .mean()
         ) < 0.001  # TODO: pick a smarter number?
-
         # address details do not change if address_id does not change
         pd.testing.assert_frame_equal(
             before_units[~mask_moved_units], after_units[~mask_moved_units]
         )
 
-        # TODO when puma/state is implemented
-        # # TODO Check that *most* are in a new state when we add all locations
-        # # Some movers are in a new state.
-        # # assert any(before.loc[mask_movers, "state"] != after.loc[mask_movers, "state"])
-
-        # # Most movers are in a different puma
-        # assert (
-        #     before.loc[mask_movers, "puma"] != after.loc[mask_movers, "puma"]
-        # ).sum() / mask_movers.sum() >= 0.95
+        # TODO Check that *most* are in a new state when we add all locations
+        # Some movers are in a new state.
+        # assert any(before.loc[mask_movers, "state"] != after.loc[mask_movers, "state"])
+        # Most movers are in a different PUMA
+        puma_col = [x for x in state_puma_cols if "puma" in x]
+        if puma_col:  # TODO: remove conditioinal when puma/state implemented for businesses
+            puma_col = puma_col[0]
```

Comment:
This is a bit odd; maybe just do this "most" check using **all** the `state_puma_cols` and do the "some" check on each individual column in `state_puma_cols`. Then you don't have to explicitly split this apart.

Reply:
ah I see what you mean